### PR TITLE
Sanitize filenames and handle member issues in `attendance.py` example script

### PR DESCRIPTION
**Pull request by alsp80**
Hi,

I came across two issues when using your project. One is about sanitising filenames based on events (for attendance) where certain special characters allowed by Spond in event names (e.g. a forward slash) would lead to unexpected behaviour or problems. I've added some code to handle these (however this might need to be reviewed for copyright as I found it somewhere but it was mentioned that it might have been taken from Django).

The other change is to handle the situation when member information is not available which lead to a crash. I'm not sure what is wrong in the account on Spond but I've included some handling for the situation when the member based on ID cannot be found. In that case the output will simply be the member guid.

The last (or actually first) commit (adding a folder into .gitignore) can probably be ignored, however I did not manage to exclude it from the Pull Request. Seems I need to beef up my git skills.

Hope some of it is useful, thanks for sharing your project, it has helped me already and I might add additional functionality and send more pull requests if ok for you.

Thanks,
Alex

### CODE REVIEW

**Comment by Olen** [comment]
Hi, and thanks for your effort.

Ideally, this should have been two PRs, but I totally understand you.
Regarding the Excel folder, as long as you don't `git add Excel/` it should not be added, but I have no problem with the gitignore.

The filename-sanitising is fine.  Even if you found it somewhere, it looks so generic that I can't see any copyright problems, even if you did not invent the exact code yourself,  It could also be that somene else here have a few comments, and maybe also have other changes they might want to add to it, so I'll leave that for now.

I'll let others also have a look at it, but at first glance, it looks fine to me.

File: attendance.py [hunk 1]
```diff
@@ -38,8 +39,11 @@ async def main():
         os.makedirs("./exports")
 
     for e in events:
+        filename = f"{e['startTimestamp']}-{e['heading']}.csv"
```
Comment (elliot-100) [inline attendance.py:42]: Would be nice to refcator this out into self-documenting function, e.g. `santised_filename(event)`

File: attendance.py [hunk 2]
```diff
@@ -50,8 +54,12 @@ async def main():
...
             )
             for o in e["owners"]:
-                person = await s.get_person(o["id"])
-                full_name = person["firstName"] + " " + person["lastName"]
+                try:
+                    person = await s.get_person(o["id"])                    
+                except KeyError:
+                    full_name = o["id"]
+                else:
+                    full_name = person["firstName"] + " " + person["lastName"]
```
Comment (elliot-100) [inline attendance.py:62]: Recommend refactoring out this repeated line to a function

File: attendance.py [hunk 3]
```diff
@@ -64,8 +72,12 @@ async def main():
                 )
             if args.a is True:
                 for r in e["responses"]["acceptedIds"]:
-                    person = await s.get_person(r)
-                    full_name = person["firstName"] + " " + person["lastName"]
+                    try:
```
Comment (elliot-100) [inline attendance.py:75]: ..and this repeated block too.

**Comment by elliot-100** [comment]
Hi - thanks for the PR! I have made a couple of comments on the code and have a couple more general points:

- It would be good to add a comment at the top of the file summarising what it does.

- The checks are failing - it looks like you have trailing whitespace. Can you run `black .` to auto-format please?

**Comment by elliot-100** [comment]
Hey @alsp80 - I hope I didn't put you off?

I'm happy to assist or resolve any of these points. Let us know... thanks again!

**Comment by elliot-100** [comment]
Thanks again @alsp80, merged!

I did do a couple of simple refactors. (I did notice some pre-existing issues here, e.g. 'organiser' column actually contains organisers or members if you use the `-a` flag, but that's one for another PR. Also I don't think the file handling in aync context is best practice).

Could you let me know if all is OK and raise a new Bug issue if not? Thanks.

**Comment by alsp80** [comment]
Hi @elliot-100,

sorry for never coming back to this thread and being silent all the time. First: thanks for accepting the pull, taking care and adapting it to how it should be. And even more so for still mentioning me a contributor.

I had always planned to come back to this and help with it but as life goes I never found the time to do so. I'm still using this project and I would have ideas and feature requests which I planned implementing but again didn't manage yet. I might add them as issues here simply to keep track of them. If I do find the time I would give it a try and see whether I can come up with another PR.

Thanks for being active on this project and offering a nice tool to do help a bit with managing groups in Spond.

**Pull request merged**